tests/Services/Search/ParametersHandlerTest.php

```diff
@@ -0,0 +1,318 @@
+                    'item.score' => 'Ceres::Template.itemRelevance'
+                ]
+            ],
+            '$isSearch is true and configSortingData does not contain item.score' => [
+                true,
+                [
+                    'default.recommended_sorting',
+                    'texts.name1_asc',
+                    'texts.name1_desc',
+                    'sorting.price.avg_asc',
+                    'sorting.price.avg_desc',
+                    'variation.createdAt_desc',
+                    'variation.createdAt_asc',
+                    'variation.availability.averageDays_asc',
+                    'variation.availability.averageDays_desc',
+                    'variation.number_asc',
+                    'variation.number_desc',
+                    'variation.updatedAt_asc',
+                    'variation.updatedAt_desc',
+                    'item.manufacturer.externalName_asc',
+                    'item.manufacturer.externalName_desc',
+                    'variation.position_asc',
+                    'variation.position_desc',
+                    'item.random'
+                ],
+                [
+                    'texts.name1_asc' => 'Ceres::Template.itemName_asc',
+                    'texts.name1_desc' => 'Ceres::Template.itemName_desc',
+                    'sorting.price.avg_asc' => 'Ceres::Template.itemPrice_asc',
+                    'sorting.price.avg_desc' => 'Ceres::Template.itemPrice_desc',
+                    'variation.createdAt_desc' => 'Ceres::Template.itemVariationCreateTimestamp_desc',
+                    'variation.createdAt_asc' => 'Ceres::Template.itemVariationCreateTimestamp_asc',
+                    'variation.position_asc' => 'Ceres::Template.itemVariationTopseller_asc',
+                    'variation.position_desc' => 'Ceres::Template.itemVariationTopseller_desc',
+                    'item.score' => 'Ceres::Template.itemRelevance'
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerGetSortingOptions
+     *
+     * @param bool $isSearch
+     * @param array $ceresConfigData
+     * @param array $expectedResult
+     */
+    public function testGetSortingOptions($isSearch, $ceresConfigData, $expectedResult)
+    {
+        $config = $this->createCeresConfigObject($ceresConfigData);
+
+        /** @var ParametersHandler|MockObject $parametersHandlerMock */
+        $parametersHandlerMock = $this->getMockBuilder(ParametersHandler::class)
+            ->setMethods(['getConfig'])
+            ->getMock();
+        $parametersHandlerMock->expects($this->once())->method('getConfig')->willReturn($config);
+
+        $this->assertEquals($parametersHandlerMock->getSortingOptions($isSearch), $expectedResult, true);
+    }
+
+    public function providerHandlePaginationAndSorting()
+    {
+        return [
+            '$isSearch is true and defaultSortingSearch is texts.name1_asc' => [
```

Comment:
Use these scenario descriptions:
* Sorting options for search page with supported default
* Sorting options for category page with supported default
* Sorting options for category page with unsupported default